src/KeyVault/KeyVault/Commands/NewAzureKeyVault.cs

```diff
@@ -97,6 +97,11 @@ public class NewAzureKeyVault : KeyVaultManagementCmdletBase
         [ValidateNotNullOrEmpty]
         public int SoftDeleteRetentionInDays { get; set; }
 
+        [Parameter(Mandatory = false,
+            HelpMessage = "Specifies whether the vault will accept traffic from public internet. If set to 'disabled' all traffic except private endpoint traffic and that that originates from trusted services will be blocked. This will override the set firewall rules, meaning that even if the firewall rules are present we will not honor the rules. By default, we will enable public network access.")]
+        [PSArgumentCompleter("Enabled", "Disabled")]
```

Comment:
Should this be a ValidateSet instead of ArgumentCompleter? Also, would it be better to use just boolean or switch parameter?

Reply:
The reason why use string in this case is PublicNetworkAccess defines as string in swagger. I'm afraid that there may be other states for PublicNetworkAccess like LimitedEnabled or sth else. So I decided to make PublicNetworkAccess as string to align swagger. What's your thought? Refer to https://github.com/Azure/azure-rest-api-specs/blob/5575aeff7364faef03c26a3661e1e64a6bb693a7/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2021-10-01/keyvault.json#L1308

Reply:
Very well

Reply:
Wanna keep ArgumentCompleter to avoid modifying validate set if adding additional valid values.